src/main/java/Ebot.java

```diff
@@ -0,0 +1,256 @@
+
+            // if user types any of the task types keyword command
+            } else if (userInput.toLowerCase().contains("todo")
+                    || userInput.toLowerCase().contains("event")
+                    || userInput.toLowerCase().contains("deadline")) {
+                handleTaskAdding(taskList, userInput);
+                writeToFile(taskList.toString());
+
+            // if user types "delete" command
+            } else if (userInput.toLowerCase().contains("delete")) {
+                handleTaskDeletion(taskList, userInput);
+                writeToFile(taskList.toString());
+
+            // if user types anything else
+            } else {
+                System.out.println(SMALL_INDENT + userInput);
+                System.out.println(LINE_DIVIDER);
+            }
+        }
+    }
+
+    private static void handleTaskDeletion(TaskList taskList, String userInput) {
+        try {
+            int deleteIndex = ((Integer.parseInt(userInput.toLowerCase().replace("delete", "").replace(" ", "")))-1);
+            taskList.deleteEntry(deleteIndex);
+            System.out.println(SMALL_INDENT + "Got it! I deleted task " + (deleteIndex + 1));
+            System.out.println(taskList);
+        } catch (NumberFormatException E) {
+            System.out.println(SMALL_INDENT + "Error! Did you forget to indicate which index to delete?");
+        } catch (IndexOutOfBoundsException E) {
+            System.out.println(SMALL_INDENT + "Task not found");
+        } finally {
+            System.out.println(LINE_DIVIDER);
+        }
+    }
+
+    private static void writeToFile(String textToAdd) {
+        try {
+            FileWriter fw = new FileWriter(OUTPUT_FILE_NAME);
+            fw.write(textToAdd);
+            fw.close();
+        } catch (IOException E) {
+            System.out.println("Unable to write to output file: " + E.getMessage());
+        }
+    }
+
+    private static void createFile() {
+        try {
+            File ebotOutput = new File(OUTPUT_FILE_NAME);
+            if ((ebotOutput.createNewFile()) && (ebotOutput.getParentFile().mkdirs())) {
+                System.out.println("Output file created: " + ebotOutput.getName());
+            } else {
+                System.out.println("Output file already exists.");
+            }
+        } catch (IOException e) {
+            System.out.println("An error occurred");
+        }
+    }
+
+    private static void handleTaskAdding(TaskList taskList, String userInput) {
+        String cleanedUserInput = userInput.toLowerCase().replace(" ", "").replace("/", "");
+        String taskDescription;
+        Task newTask;
+        if (userInput.toLowerCase().contains("todo")) {
+            taskDescription = cleanedUserInput.replace(("todo"), "");
+            try {
+                if (taskDescription.length() == 0) {
+                    throw new MissingDescriptionException();
+                }
+                newTask = new Todo(taskDescription, false);
```

Comment:
this is 3 levels of indentation, consider refactoring inner lines of code to another method